File: 1446.py
class Solution:
    def maxPower(self, s: str) -> int:
        ans = 1
        count = 1
        for i in range(1, len(s)):
            if s[i] == s[i-1]:
                count += 1
                # ans = max(ans, count)
            else:
                ans = max(ans, count)
                count = 1
        return max(ans, count)


if __name__ == "__main__":
    s = Solution()
    print(s.maxPower('abbcccddddeeeeedcba'))

File: 1446.go
package leetcode

func maxPower(s string) int {
	// ans := 0
	// l := 0
	// for r := 1; r < len(s); r++ {
	//     if s[r] != s[l]{
	//         if tmp := r - l; tmp > ans {
	//             ans = tmp
	//         }
	//         l = r
	//     }
	// }
	// if tmp := len(s) - l; tmp > ans {
	//     ans = tmp
	// }
	// return ans
	var (
		ans = 0
		cnt = 1
	)
	for i := 1; i < len(s); i++ {
		if s[i] != s[i-1] {
			if cnt > ans {
				ans = cnt
			}
			cnt = 1
		} else {
			cnt++
		}
	}
	if cnt > ans {
		ans = cnt
	}
	return ans
}
